impl/impl.go

```diff
@@ -189,6 +194,17 @@ func (m *manager) OpenPushDataChannel(ctx context.Context, requestTo peer.ID, vo
 	if err != nil {
 		return chid, err
 	}
+
+	eventsCb := tc.EventsCb()
+	if eventsCb != nil {
+		m.channelSubscriptions.Subscribe(chid, eventsCb)
+	}
+
+	err = m.channels.Open(chid)
+	if err != nil {
+		return chid, err
+	}
```

Comment:
not necessary, just seeing an opportunity

```suggestion
	if eventsCb := tc.EventsCb(); eventsCb != nil {
		m.channelSubscriptions.Subscribe(chid, eventsCb)
	}

	if err := m.channels.Open(chid); err != nil {
		return chid, err
	}
```